### PR TITLE
Update go-libp2p to v0.15.0

### PR DESCRIPTION
**Pull request by aarshkshah1992**
`go-libp2p-core` v0.9.0 and `go-libp2p` v0.15.0 contains breaking changes.

### CODE REVIEW

**Comment by jennijuju** [comment]
Ci is very unhappy..
What’s the breaking changes? What are the potential  impact to lotus users? Have we tested it?

File: go.mod [hunk 1]
```diff
@@ -85,7 +85,7 @@ require (
 	github.com/ipfs/go-ipfs-exchange-interface v0.0.1
 	github.com/ipfs/go-ipfs-exchange-offline v0.0.1
 	github.com/ipfs/go-ipfs-files v0.0.8
-	github.com/ipfs/go-ipfs-http-client v0.0.5
+	github.com/ipfs/go-ipfs-http-client v0.1.0-update-libp2p
```
Comment (Stebalien) [inline go.mod:88]: Can you make a PR for this? We might as well update go-ipfs.

**Comment by Stebalien** [comment]
>  What’s the breaking changes? What are the potential impact to lotus users? Have we tested it?

The main breaking changes are the removal of deprecated interfaces. That shouldn't affect anything.

Other than that, this has a new mDNS implementation (not used by lotus, IIRC) and a new QUIC version.

The riskiest thing is the new QUIC version.

File: go.mod (hunk 1 above)
Comment (aarshkshah1992) [reply go.mod:88]: @Stebalien Master in `github.com/ipfs/go-ipfs-http-client` already has the latest mutliaddr stuff but upgrading to it breaks a whole host of other ipfs related stuff in Lotus and I don't wanna do that upgrade as part of this PR.

Comment (Stebalien) [reply go.mod:88]: Can we tag v0.0.6 then?

File: node/modules/lp2p/addrs.go [hunk 2]
```diff
@@ -11,19 +11,6 @@ import (
 	mamask "github.com/whyrusleeping/multiaddr-filter"
 )
 
-func AddrFilters(filters []string) func() (opts Libp2pOpts, err error) {
```
Comment (marten-seemann) [inline node/modules/lp2p/addrs.go:14]: Was this an unused function? Otherwise, you might need to replace it with a connection gater.

**Review by marten-seemann — APPROVED** [review]

File: node/modules/lp2p/addrs.go (hunk 2 above)
Comment (aarshkshah1992) [reply node/modules/lp2p/addrs.go:14]: Yeah, it was dead code.

File: go.mod (hunk 1 above)
Comment (aarshkshah1992) [reply go.mod]: @Stebalien  This is done. Please can you approve the PR ?

File: testplans/lotus-soup/go.mod [hunk 3]
```diff
@@ -8,30 +8,30 @@ require (
...
 	github.com/ipfs/go-cid v0.1.0
-	github.com/ipfs/go-datastore v0.4.5
+	github.com/ipfs/go-datastore v0.4.6
 	github.com/ipfs/go-ipfs-files v0.0.8
 	github.com/ipfs/go-ipld-format v0.2.0
 	github.com/ipfs/go-log/v2 v2.3.0
 	github.com/ipfs/go-merkledag v0.3.2
 	github.com/ipfs/go-unixfs v0.2.6
-	github.com/ipld/go-car v0.1.1-0.20201119040415-11b6074b6d4d
+	github.com/ipld/go-car v0.3.1-0.20210601190600-f512dac51e8e
```
Comment (Stebalien) [inline testplans/lotus-soup/go.mod:29]: This is not a commit on master, or a tagged commit.

**Review by Stebalien — CHANGES_REQUESTED** [review]

File: testplans/lotus-soup/go.mod (hunk 3 above)
Comment (aarshkshah1992) [reply testplans/lotus-soup/go.mod:29]: @Stebalien This has been a problem with the `go-car` dep for a while now. Do you just want us to tag this commit and push it in ?

Comment (Stebalien) [reply testplans/lotus-soup/go.mod:29]: That works. As long as we can be sure it won't disappear.

**Comment by dirkmc** [comment]
I ran storage and retrieval tests between a client and miner running either
- v1.11.3-rc2
- v1.11.3-rc2 + commits on this branch: "hybrid"

### Results
- hybrid client vs hybrid miner
  - Storage deal success
  - Retrieval deal success

- v1.11.3-rc2 client vs hybrid miner
  - Storage deal success
  - Retrieval deal success

- hybrid client vs v1.11.3-rc2 miner
  - Storage deal success
  - Retrieval deal success

File: testplans/lotus-soup/go.mod (hunk 3 above)
Comment (aarshkshah1992) [reply testplans/lotus-soup/go.mod]: @Stebalien - This is done. Please can you approve ?

**Review by magik6k — APPROVED** [review]

**Pull request merged**